src/CRUD.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable indent */
+/* eslint-disable comma-dangle */
+/* eslint-disable quotes */
+/* eslint-disable no-bitwise */
+/* eslint-disable no-use-before-define */
+/* eslint-disable no-unused-expressions */
+// select element
+
+const form = document.getElementById("todoform");
+const todoInput = document.getElementById("newtodo");
+const todoListElement = document.getElementById("tododslist");
+
+// My variables
+let todos = JSON.parse(localStorage.getItem("todos")) || [];
+let editToDoId = -1;
+
+// First render
+renderTodos();
+
+// To submit form
+form.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  saveTodo();
+  renderTodos();
+  localStorage.setItem("todos", JSON.stringify(todos));
+});
+
+// to save to do
+function saveTodo() {
+  const todoValue = todoInput.value;
+
+  // check if empty
+  const isEmpty = todoValue === "";
+
+  //  check for duplicates
+  const isDuplicate = todos.some(
+    (todo) => todo.value.toUpperCase() === todoValue.toUpperCase()
+  );
+  if (isEmpty) {
+    alert("Todo's input is empty");
+  } else if (isDuplicate) {
+    alert("To do exists");
+  } else {
+    if (editToDoId >= 0) {
+      // update the edited to do
+      todos = todos.map((todo, index) => ({
+        ...todo,
+        value: index === editToDoId ? todoValue : todo.value,
+      }));
+      editToDoId = -1;
+    } else {
+      todos.push({
+        value: todoValue,
+        checked: false,
+        color: `#${(((1 << 24) * Math.random()) | 0)
+          .toString(16)
+          .padStart(6, "0")}`,
+      });
```

Comment:
- [ ] Kindly make sure By default new tasks have the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).

- [ ] Kindl make sure by deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1).

     **_I highly recommend to implement in order to meet the project's requirements_**